### PR TITLE
add beginner's guide to the airbytecatalog

### PR DESCRIPTION
**Pull request by cgardens**
## What
* Had some time on the plane to take a pass at a beginner's guide to the AirbyteCatalog. The goal is to provide a doc to be used by a first time contributor (someone following the toy connector guide (or equivalent))
* Everyone I have worked with on building a new connector has been confused about the catalog (or at least about how to use it).

## Open Questions
* Should this replace the existing catalog docs? I'm inclined to say no. The existing docs are more detailed and helpful for someone with more experience.

### CODE REVIEW

File: docs/tutorials/beginners-guide-to-catalog.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+Let's walk through what each field in a stream means.
+* `name` - The name of the stream.
+* `supported_syn_modes` - This field lists the type of data replication that this source supports. The possible values in this array include `FULL_REFRESH` ([docs](../architecture/full-refresh.md)) and `INCREMENTAL` ([docs](../architecture/incremental.md)).
+* `source_defined_cursor` - If the stream supports `INCREMENTAL` replication, then this field signal whether the source can figure out how to detect new records on its own or not.
+* `json_schema` - This field is a [JsonSchema](https://json-schema.org/understanding-json-schema) object that describes the structure of the data. Notice that each key in the `properties` object corresponds to a column name in our database table.
+
+Now we understand _what_ data is available from this source. Next we will configure _how_ we want to replicate that data.
+
+### ConfiguredAirbyteCatalog
+Let's say that we do not care about replicating the pilot data at all. We do want to replicate the airlines data as a `FULL_REFRESH`. Here's what our `ConfiguredaAirbyteCatalog` would look like.
```
Comment (jrhizor) [inline docs/tutorials/beginners-guide-to-catalog.md:98]:
```suggestion
Let's say that we do not care about replicating the pilot data at all. We do want to replicate the airlines data as a `FULL_REFRESH`. Here's what our `ConfiguredAirbyteCatalog` would look like.
```

**Review by jrhizor — APPROVED** [review]
This guide looks helpful.

However, it reinforces my belief that we should output configured catalogs by default from `discover` so 99.99% of users and contributors won't need to think about this at all.

File: docs/tutorials/beginners-guide-to-catalog.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+      }
+    }
+  ]
+}
+```
+
+Just as with the `AirbyteCatalog` the `ConfiguredAirbyteCatalog` contains a list. This time it is a list of `ConfiguredAirbyteStream`s (instead of just `AirbyteStream`s).
+
+Let's walk through each field in the `ConfiguredAirbyteStream`:
+* `sync_mode` - This field must be one of the values that was in `supported_sync_modes` in the `AirbyteStram` - Configures which sync mode will be used when data is replicated.
```
Comment (michel-tricot) [inline docs/tutorials/beginners-guide-to-catalog.md:132]: Typo AirbyteStram -> AirbyteStream

**Review by michel-tricot — COMMENTED** [review]
Looks great!

**Pull request merged**